### ISSUE TITLE
Minor Visual Flickering Bug

### ISSUE THREAD

**Issue by Ghostrider231** [Electroblob77/Wizardry#12]
Minecraft version: 1.12.2
Wizardry version: 4.1.0

Issue details: Trying out the mod today, while messing around with some wizard vs evil wizard/wizard vs evoker. Sometimes, or most the time whenever one of the wizards cast a spell that involves particle effects: the water, rain, or other particles would flicker as they were in complete darkness.

Repeating the issue:

1. Have some wizards fight some random mobs with water/rain in sight. (Spawn eggs work too)
2. Watch the wizards fight other mobs while some water/rain is in sight.
3. The water may flicker as soon as any of the wizards cast a spell.

Other mods are totally out of the question, I've tried this only with this and it still happens. This is only a Minor Issue but is kind of annoying.

**Comment by Electroblob77** [comment]
I can't seem to reproduce this effect, would you mind trying to find out which spells are causing this?

**Comment by Ghostrider231** [comment]
Just tested again. I've also tried spawning plentiful of wizards and plentiful of mobs for them to fight. I also tried to do this while near to a body of water. The spells that can/are likely triggering the flicker are (Handful of them that I know):

------------

Greater Fireball
Forcefield (I don't know if this makes it happen more)
Frost Ray (Interesting enough, the player can trigger this one)
Smoke bomb
Fire bomb
Poison bomb
Detonate
Wither Skull

Smoke Bomb and Fire Bomb are apparently the greatest offender. Cause the fact I'm able to trigger it myself as well. (Happens more often if a wizard or more is around)

----------------

===

**Edit: Much better way of reproducing the problem (This can be done by the player):**

1. Be in a area where it could rain or snow. (I've tested snowy areas but not rainy areas)
2. use the "/weather rain" command. Without quotes.
3. Spawn a mob, then cast a Frost Ray Spell at it.
4. Equip a sword, then attack the mob inflicted by the Frost Ray Slowness. (Sword needs to have the swish animation for this to happen)
5. Observe as the rain/snow effects flickers or blinks as you just attacked the mob. (Or as the swish animation shows)
6. If it doesn't flicker, Repeat to Step 3 If Possible. (If the game decided to stop the weather, repeat to step 2)

Apparently killing the mob with the frost ray particle effects deters the bug temporary, but returns If I inflict another mob with the same spell as well. Which gives the Frost Ray particle effects on the mob again.

[GIF of what I mean](https://imgur.com/4ZT3n9w) 

(Don't worry about the command block, It doesn't have anything to do with this. It's only there to disable slime spawning)

**If I have Optifine installed and using it's internal shaders. The water won't flicker as much while the others still. BUT having Optifine installed brings up this problem**: https://github.com/Electroblob77/Wizardry/issues/15
